ggml-vulkan.cpp

```diff
@@ -978,8 +978,10 @@ void ggml_vk_init() {
 #ifdef GGML_VULKAN_VALIDATE
         "VK_EXT_validation_features",
 #endif
+        "VK_KHR_portability_enumeration",
     };
-    vk::InstanceCreateInfo instance_create_info(vk::InstanceCreateFlags(), &app_info, layers, extensions);
+
+    vk::InstanceCreateInfo instance_create_info(vk::InstanceCreateFlags(vk::InstanceCreateFlagBits::eEnumeratePortabilityKHR), &app_info, layers, extensions);
```

Comment:
I'm gonna do a proper test tomorrow. But you should wrap these changes into preprocessor checks so that they only happen on Apple devices.

Reply:
I considered that, but then again if these features do not cause problems on other platforms, and it yields a more generic and simple solution, shouldn't we just leave them on?

Your call. Curious to see whether you're running into any issues on your side.

Reply:
Thanks for the super quick feedback by the way. ❤️

Reply:
> I considered that, but then again if these features do not cause problems on other platforms, and it yields a more generic and simple solution, shouldn't we just leave them on?
> 
> Your call. Curious to see whether you're running into any issues on your side.

It doesn't cause any issues for me, but it might on other systems. What you're using is an instance extension, and it will not be available on all devices. There's also no point enabling it on any other system, since the Vulkan drivers on Linux and Windows are conformant, unlike MoltenVK.

Even on Mac the correct way is checking whether that extension is available before trying to load it. Do you want to add these checks yourself or would you prefer me to do that?

Reply:
I don't have the foggiest how to add these checks and don't really have bandwitdh to figure it out, honestly.

But will happily wrap my patch with `#if defined (__APPLE__)` (not sure if that's the appropriate but that's what I found).

Very very very happy with the results by the way. Very stable, I now generate without thermal throttling, which promises more gains in the near future! :)

Reply:
I see the exact patch necessary seems to be in #5322. Happy to blend the patches and update in the next couple of days. Or you can just merge that one if you'd like to move ahead.

Reply:
Please do, yeah. If you're not sure how to check for the instance availability, I'll add that part after you did the Mac checks.